File: Program/Java Program/Package  Program/Employee_Package/src/staff/Employee.java
package staff;

public abstract class Employee
{
    public String name;
    public int id;
    public double salary;

    public void setName(String s)
    {
        this.name=s;
    }
    public  String getName()
    {
        return this.name;

    }
    public void setId(int a)
    {
        this.id = a;
    }
    public int getId()
    {
        return this.id;

    }
    public  void setSalary(double sa)
    {
        this.salary = sa;
    }

    public double  getSalary()
    {
        return this.salary;
    }

    public abstract  double calSalary();
   /*  {
          return this.salary;
      }*/


    public  Employee()
    {
        this.name="Not given";
        this.id = 0;
        this.salary = 0;


    }

    public Employee(String s, int a, double sa)
    {
        this.name=s;
        this.id = a;
        this.salary = sa;


    }

    public void display()
    {
        System.out.println("Name of Employee is: "+this.name);
        System.out.println("ID of Employee is: "+this.id);
        System.out.println("Salary of Employee is: "+this.salary);

    }
}//class employee ends here

File: Program/C++ prog/Employee.cpp
#include <stdio.h>
#include <string.h>

class Emp
{
    int id;
    char name[20];
    float sal;

  public:
	  Emp() //Default constructor
	  {
		  printf("\n Default constructor of Employe call");
		  //cout << constuctor of employe call;
		  id = 100;
		  strcpy(name, "Not Given");
		  salary = 10000;
	  }
	  Emp(int i, char*nm, double s) //Parameterized Constructor
		  ////Pass the parameter for parameterised constructor
	  {
		  printf("\n parametrised constructor of Employe call");
		  id = i;
		  strcpy(name, nm);
		  salary = s;


	  }
    void setId(int a)
    {
        this->id = a;
    }
    void setName(char *nm)
    {
        strcpy(this->name, nm);
    }

    void setSal(float b)
    {
        this->sal = b;
    }

    int getId()
    {
        return this->id;
    }
    char *getName()
    {
        return this->name;
    }
    float getSal()
    {
        return this->sal;
    }
    void display()
    {
        printf("Id is:%d Name is:%s Sal is :%ld", this->id, this->name, this->sal);
    }
	friend ostream & operator << (ostream &, Employe &);
	friend istream& operator>>(istream&, Employe &);
};
ostream & operator << (ostream &o, Emp &e)
{
	o << e.getId() << e.getName() << e.getsalary() << endl;
	return o;
}
istream& operator>>(istream& i, Emp &e)
{
	i >> e.id >> e.name >> e.salary;
	return i;
}

int main()
{
    Emp e1;
	cin >> e1;

   
}
